pkg/agent/monitortool/latency_store.go

```diff
@@ -0,0 +1,242 @@
+// Copyright 2024 Antrea Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package monitortool
+
+import (
+	"errors"
+	"net"
+	"sync"
+	"time"
+
+	"github.com/containernetworking/plugins/pkg/ip"
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/klog/v2"
+
+	"antrea.io/antrea/pkg/util/k8s"
+)
+
+// LatencyStore is a store for latency information of connections between nodes.
+type LatencyStore struct {
+	// Lock for the latency store
+	mutex sync.RWMutex
+
+	// Whether the agent is running in network policy only mode
+	isNetworkPolicyOnly bool
+	// The map of node ip to latency entry, it will be changed by latency monitor
+	nodeIPLatencyMap map[string]*NodeIPLatencyEntry
+	// The map of node ip to node name, it will be changed by node watcher
```

Comment:
The comment is not accurate, given that we are mapping from node name to node IPs (and not the reverse).
I do also think that we should rename the field, because in networkPolicyOnly mode, we are not using the _gateway_ IP. So maybe `nodeTargetIPsMap` would be a better name?

Reply:
Sounds good, there is no modification to the definition of the original code when adding new logic, which I have modified.